module/zfs/dsl_synctask.c

```diff
@@ -85,6 +85,11 @@ dsl_sync_task_common(const char *pool, dsl_checkfunc_t *checkfunc,
 
 	dmu_tx_commit(tx);
 
+	if (sigfunc != NULL && txg_wait_synced_sig(dp, dst.dst_txg)) {
+		/* current contract is to call func once */
+		sigfunc(arg, tx);
+		sigfunc = NULL;	/* in case of an EAGAIN retry */
```

Comment:
I'm not sure I understand the comment here.  It doesn't appear the is any special handling for EAGAIN here.

Reply:
We only want to run the `sigfunc` once. So we clear it here in case an EAGAIN causes us to loop around.  I'll clean up the comment to make it more obvious.